### DISCUSSION TITLE
Problems too install - Modules installed outside the virtual environment were found.

### DISCUSSION THREAD

**Discussion by ticannas** [bmaltais/kohya_ss#921] [Q&A]
Heey, im not one dev guy, dont know much about those pip install things, i actually work with art
can somebody help me with this please"

Modules installed outside the virtual environment were found.
This can cause issues. Please review the installed modules.

You can uninstall all local modules with:

←[1;34mdeactivate
pip freeze > uninstall.txt
pip uninstall -y -r uninstall.txt
←[1;33m=============================================================←[0m

Traceback (most recent call last):
  File "D:\Local A.I. Lab\LoRA - Low Rank Adaptations Training\kohya_ss\tools\validate_requirements.py", line 6, in <module>
    from library.custom_logging import setup_logging
ModuleNotFoundError: No module named 'library'

addicioanally i did installed those CUDNN 8.6 for my 3050

**Comment by Trojaner** [comment]
I am facing the same issue

**Comment by Mistborn-First-Era** [comment]
Just updated and go the same error.

**Reply by Mistborn-First-Era** [reply]
Found a temporary solution. Not sure if the first step is 100% needed so do at your own risk.
1. go to your VENV folder in terminal. and then run `pip freeze > uninstall.txt` then do `pip uninstall -y -r uninstall.txt` from the same location
2. Download and install [THIS](https://github.com/bmaltais/kohya_ss/releases/tag/v21.5.14) old release. 
3. Don't update, just run the setup. 
Should be usable for now.

**Comment by Omarkh12** [comment]
im having the same error has anyone found an answer yet?

**Comment by tpcdaz** [comment]
I really wish these developers would stop changing stuff when it works. This new setup file is a disaster. Before it just worked. Now the dev has tried to be too clever and broke everything ffs

**Comment by UltimatePractitioner** [comment]
Same problem here.

**Comment by ticannas** [comment]
its really hard being a Art guy in those pip install things hahahahaha im afraid ill be pushed into becaming a amish hahahahaha

**Comment by 2blackbar** [comment]
What up devs ? what was the last working comit ? ITs 2 weeks since this repo is unuseable

**Comment by primeinc** [comment]
Hey - quick fix here, delete or comment out the following in [tools/validate_requirements.py](https://github.com/bmaltais/kohya_ss/blob/c0eb172958dfed3522211682a6a6bb0ea1f69869/tools/validate_requirements.py#LL6C1-L9C22) [lines 6-9]
```
from library.custom_logging import setup_logging

# Set up logging
log = setup_logging()
```
replace it with
```
import logging

# Set up logging
logging.basicConfig(level=logging.INFO)

# Create a logger
log = logging.getLogger(__name__)
```

**Reply by m3diaoffline** [reply]
You are our hero, thank you, this works

**Reply by jonbjornn** [reply]
Still appreciated - fix works as of August 25th. Not sure if this is still needed for a fresh install but I've had Kohya installed for several months and recently updated. Kept crashing out till I made this edit

**Reply by Mesalinc** [reply]
Did this on a fresh install and still no dice...

**Comment by hansolocambo** [comment]
[primeinc](https://github.com/primeinc)'s solution works. Thanks. 

P.S : "**validate_requirements.py**" is not in _tools_ but in the _setup_ folder.

**Comment by AI-Underdog** [comment]
keep getting this error ModuleNotFoundError: No module named 'library'

**Reply by rpfile** [reply]
@AI-Underdog for some reason the module search path isn't being set right somewhere in the code. if you know the python file where that error is occurring, right before the import command for library (near the top of the file) you can add a couple of lines of code to add the search path. there are a number of posts in the issue section with the lines of code you need. just search for library in the issues area.

**Comment by oojar** [comment]
This error also happens when changing the location of kohya_ss, The path of the installation is embedded in activate.bat.
Edit the activate.bat in ...\kohya_ss\venv\Scripts
Change 'set VIRTUAL_ENV=' to match the current location of the kohya_ss directory.

Thanks to thiagopasseri for pointing this out!

**Reply by rpfile** [reply]
thanks - that was exactly what i did. i like to keep different versions around and change the name of the kohya_ss directory to add a version number. so that must have caused the problem.

**Reply by DarkAlchy** [reply]
set VIRTUAL_ENV=F:\kohya_ss-win\venv

That is precisely where it is located, so that isn't it for me having this same message.
